likeminds-chat-reactnative-ui/components/commonFunctions/index.tsx

```diff
@@ -0,0 +1,188 @@
+                  if (!!!isLongPress) {
+                    const urlRegex = /(https?:\/\/[^\s]+)/gi;
+                    const emailRegex =
+                      /\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,}\b/g;
+                    let isURL = urlRegex.test(val);
+                    let isEmail = emailRegex.test(val);
+
+                    if (isEmail) {
+                      await Linking.openURL(`mailto:${val}`);
+                    } else if (isURL) {
+                      await Linking.openURL(val);
+                    } else {
+                      await Linking.openURL(`https://${val}`);
+                    }
+                  }
+                }}
+              >
+                <Text
+                  style={{
+                    color: Styles.$COLORS.LIGHT_BLUE,
+                    fontSize: Styles.$FONT_SIZES.MEDIUM,
+                    fontFamily: Styles.$FONT_TYPES.LIGHT,
+                  }}
+                >
+                  {val}
+                </Text>
+              </Text>
+            ) : (
+              <Text>{val}</Text>
+            )}
+          </Text>
+        ))}
+      </Text>
+    );
+  } else {
+    return message;
+  }
+}
+
+// naruto: naruto|route://member_profile/88226?member_id=__id__&community_id=__community__>>
+// test string = '<<Sanjay kumar 🤖|route://member/1260>> <<Ishaan Jain|route://member/1003>> Hey google.com';
+// This decode function helps us to decode tagged messages like the above test string in to readable format.
+// This function has two responses: one for Homefeed screen and other is for chat screen(Pressable ones are for chat screen).
+// The REGEX_USER_SPLITTING is used to split the text into different parts based on the regex specified and then using a for loop tags are shown differently along with name and route
+export const decode = (
+  text: string | undefined,
+  enableClick: boolean,
+  chatroomName?: string,
+  communityId?: string,
+  isLongPress?: boolean,
+  memberUuid?: string,
+  chatroomWithUserUuid?: string,
+  chatroomWithUserMemberId?: string
+) => {
+  if (!text) {
+    return;
+  }
+  let arr: any[] = [];
+  let parts = text?.split(REGEX_USER_SPLITTING);
+
+  if (!!parts) {
+    for (const matchResult of parts) {
+      if (!!matchResult.match(REGEX_USER_TAGGING)) {
+        let match = REGEX_USER_TAGGING.exec(matchResult);
+        if (match !== null) {
+          let { name, route } = match?.groups!;
+
+          const startingIndex = route.indexOf("/");
+          const taggedUserId = route.substring(startingIndex + 1);
+
+          // LMChatAnalytics.track(
+          //   Events.USER_TAGS_SOMEONE,
+          //   new Map<string, string>([
+          //     [Keys.COMMUNITY_ID, communityId?.toString()],
+          //     [Keys.CHATROOM_NAME, chatroomName?.toString()],
+          //     [Keys.TAGGED_USER_ID, taggedUserId?.toString()],
+          //     [Keys.TAGGED_USER_NAME, name?.toString()],
+          //   ]),
+          // );
```

Comment:
why is this commented

Reply:
because I think we should not have analytics in UI layer, also if it has to be here we need props like `communityId`, `chatroomName`, etc which we wont be having in UI layer so commented it out. will remove this part.

Reply:
then remove it completely. And find some way on integration layer to have these analytics